packages/eslint-plugin-typescript/src/index.ts

```diff
@@ -74,34 +77,77 @@ export const cloudflightTypescriptDisableTypeCheckedConfig = tseslint.config({
     },
 });
 
-export const cloudflightTypescriptConfig = tseslint.config(
-    // eslint-disable-next-line @typescript-eslint/no-deprecated
-    ...cloudflightTypescriptBaseConfig,
-    // eslint-disable-next-line @typescript-eslint/no-deprecated
-    ...cloudflightTypescriptImportConfig,
-    // eslint-disable-next-line @typescript-eslint/no-deprecated
-    ...cloudflightTypescriptDisableTypeCheckedConfig,
-);
+export interface CloudflightEslintPluginSettings {
+    rootDirectory: string;
+    /**
+     * Override the default tsconfig files to use for the project.
+     * Keep this list as short as possible, a large list will negatively impact performance.
+     * Relative to the rootDirectory.
+     */
+    tsConfigFiles?: string[];
```

Comment:
I am not sure about leaving this optional, maybe we should just always let the user provide the tsconfig file names. That way we wouldn't have any hidden options

Reply:
I will leave this as is for now, since we want to provide sane defaults. It would still be easy to change in the future if needed.